### PR TITLE
Add unit-test, linter, and gofmt jobs for 4.10 release for ovn-k

### PR DESCRIPTION
**Pull request by abhat**
This commit adds missing utest, linter, and gofmt jobs for ovn-k

Signed-off-by: Aniket Bhat <anbhat@redhat.com>

### CODE REVIEW

**Comment by abhat** [comment]
cc @trozet.

File: ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-release-4.10.yaml [hunk 1]
```diff
@@ -154,6 +154,27 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-e2e-aws-ovn-serial
+- as: unit
+  commands: |
+    cd go-controller
+    PKGS=$(go list -mod vendor -f '{{if len .TestGoFiles}} {{.ImportPath}} {{end}}' ${PKGS:-./cmd/... ./pkg/... ./hybrid-overlay/...} | xargs)
+    # this package needs running as a container for TestNewClientset
+    PKGS=${PKGS//"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"/ }
+    PKGS=$PKGS make test NOROOT=TRUE
```
Comment (jluhrsen) [inline ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-release-4.10.yaml:163]: are you missing a ';' after PKGS=$PKGS?

I know this is the same syntax in the other versions and you are just matching that here, but running it like this seems like a noop when I do this locally.

Comment (abhat) [reply ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-release-4.10.yaml:166]: it shouldn't be a noop. i can run it this way, make sure PKGS has an actual value and you are in the go-controller directory.

Comment (jluhrsen) [reply ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-release-4.10.yaml:166]: weird, I don't know what's wrong with my setup then. here's how it looks to me:
```

[jamoluhrsen@thinkpad go-controller]$ pwd
/home/jamoluhrsen/repos/RedHat/openshift/ovn-kubernetes-1/go-controller
[jamoluhrsen@thinkpad go-controller]$ PKGS=$(go list -mod vendor -f '{{if len .TestGoFiles}} {{.ImportPath}} {{end}}' ${PKGS:-./cmd/... ./pkg/... ./hybrid-overlay/...} | xargs)
[jamoluhrsen@thinkpad go-controller]$ echo $PKGS
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory github.com/ovn-org/ovn-kubernetes/go-controller/pkg/informer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube/healthcheck github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/unidling github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/gateway github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator/allocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/loadbalancer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/logical_switch_manager github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/subnetallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovndbmanager github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/controller
[jamoluhrsen@thinkpad go-controller]$ PKGS=${PKGS//"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"/ }
[jamoluhrsen@thinkpad go-controller]$ echo $PKGS
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory github.com/ovn-org/ovn-kubernetes/go-controller/pkg/informer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube/healthcheck github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/unidling github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/gateway github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator/allocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/loadbalancer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/logical_switch_manager github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/subnetallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovndbmanager github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/controller
[jamoluhrsen@thinkpad go-controller]$ PKGS=$PKGS make test NOROOT=TRUE
podman run -it --rm --security-opt label=disable -e NOROOT=TRUE -v /home/jamoluhrsen/repos/RedHat/openshift/ovn-kubernetes-1:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.18.4 sh -c "RACE=1 DOCKER_TEST=1 COVERALLS= PKGS="github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory github.com/ovn-org/ovn-kubernetes/go-controller/pkg/informer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube/healthcheck github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/unidling github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/gateway github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator/allocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/loadbalancer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/logical_switch_manager github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/subnetallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovndbmanager   github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/controller" hack/test-go.sh focus \"\" "
[jamoluhrsen@thinkpad go-controller]$ echo $?
0
[jamoluhrsen@thinkpad go-controller]$ time PKGS=$PKGS make test NOROOT=TRUE
podman run -it --rm --security-opt label=disable -e NOROOT=TRUE -v /home/jamoluhrsen/repos/RedHat/openshift/ovn-kubernetes-1:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.18.4 sh -c "RACE=1 DOCKER_TEST=1 COVERALLS= PKGS="github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory github.com/ovn-org/ovn-kubernetes/go-controller/pkg/informer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube/healthcheck github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/unidling github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/gateway github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator/allocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/loadbalancer github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/logical_switch_manager github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/subnetallocator github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovndbmanager   github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/controller" hack/test-go.sh focus \"\" "

real	0m0.564s
user	0m0.192s
sys	0m0.090s
[jamoluhrsen@thinkpad go-controller]$ 

```

**Pull request merged**